app/models/offer.rb

```diff
@@ -207,6 +208,34 @@ def nondraft_states
     def donor_states
       valid_states - ["draft"]
     end
+
+    def offer_active_states_counter(offer_filter_param)
+      grouped_offers = filter(offer_filter_param).group_by(&:state)
+      offers_count = offers_count_per_state(grouped_offers)
+      offers_count["offers_total_count"] = offers_count_sum(offers_count) unless offer_filter_param[:priority]
+      offers_count = prepend_offer_key_with("priority", offers_count) if offer_filter_param[:priority]
+      offers_count = prepend_offer_key_with("reviewer", offers_count) if offer_filter_param[:self_reviewer]
+      offers_count
+    end
+
+    def offers_count_for(self_reviewer: false)
+      res = {}
+      res.merge! offer_active_states_counter({ state_names: ACTIVE_OFFERS, priority: false, self_reviewer: self_reviewer })
+      res.merge! offer_active_states_counter({ state_names: ACTIVE_OFFERS, priority: true, self_reviewer: self_reviewer })
+      res
+    end
+
+    def prepend_offer_key_with(prefix, offer)
+      offer.transform_keys { |key| "#{prefix}_#{key}" }
+    end
+
+    def offers_count_sum(offer)
+      offer.values.reduce(:+)
+    end
+
+    def offers_count_per_state(offer)
```

Comment:
Quick warning on those 2 functions here

You're receiving an argument and modifying it. 
That's always a dangerous practice, we want to write pure functions as much as possible to avoid potential side effect (big part of state management).

It's fairly contained in this case, which is fine. But generally if you are not 100% sure of the safety of that operation it's better not to. (better safe than sorry)

I try to force myself to never modify any variable (as much as I can).
Our server is stateless, so applying that in our code makes sense.

Reply:
not a change request. just a topic I'm passionate about